app/controllers/course/announcements_controller.rb

```diff
@@ -51,4 +52,18 @@ def destroy #:nodoc:
   def announcement_params #:nodoc:
     params.require(:announcement).permit(:title, :content, :sticky, :valid_from, :valid_to)
   end
+
+  # Ensure that the component is enabled and load its settings.
+  #
+  # @raise [Coursemology::ComponentNotFoundError] When the component is disabled.
+  def check_component_and_load_settings
```

Comment:
I would separate this into a separate class for more complex components, and all controllers in that class would inherit from that controller.